ocs_ci/ocs/resources/storage_cluster.py

```diff
@@ -2145,3 +2175,86 @@ def wait_for_consumer_storage_provider_endpoint_in_provider_wnodes(
         func=check_consumer_storage_provider_endpoint_in_provider_wnodes,
     )
     return sample.wait_for_func_status(result=True)
+
+
+def get_storageclass_names_from_storagecluster_spec():
+    """
+    Retrieve storage class names from the storage cluster's spec.
+
+    This function queries the storage cluster's specification and returns a dictionary containing
+    the storage class names for various resources, such as cephFilesystems, cephObjectStores,
+    cephBlockPools, cephNonResilientPools, nfs, and encryption.
+
+    Returns:
+        dict: A dictionary containing the storage class names for various resources.
+            The keys are the names of the resources, and the values are the respective storage
+            class names. If a resource does not have a storage class name, it will be set to None.
+    """
+    sc_obj = ocp.OCP(
+        kind=constants.STORAGECLUSTER,
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    )
+
+    keys_to_search = [
+        constants.OCS_COMPONENTS_MAP["cephfs"],
+        constants.OCS_COMPONENTS_MAP["rgw"],
+        constants.OCS_COMPONENTS_MAP["blockpools"],
+    ]
+
+    spec_data = sc_obj.get()["items"][0]["spec"]  # Get the "spec" data once
+
+    data = {
+        key: value.get("storageClassName")
+        for key, value in spec_data.get("managedResources", {}).items()
+        if key in keys_to_search and value.get("storageClassName")
+    }
+
+    # get custom storageclass name for nonresilientPools
+    nonresilientpool_key = constants.OCS_COMPONENTS_MAP["cephnonresilentpools"]
+    nonresilientpool_data = spec_data["managedResources"].get(nonresilientpool_key, {})
+    storage_class_name = nonresilientpool_data.get("storageClassName")
+
+    if nonresilientpool_data.get("enable") and storage_class_name:
+        data[nonresilientpool_key] = storage_class_name
+
+    # Get custom storageclass name for 'nfs' service
+    if spec_data.get("nfs", {}).get("enable"):
+        nfs_storage_class_name = spec_data["nfs"].get("storageClassName")
+        if nfs_storage_class_name:
+            data["nfs"] = nfs_storage_class_name
+
+    # Get custom storageclass name for 'encryption' service
+    if spec_data.get("encryption", {}).get("enable"):
+        encryption_storage_class_name = spec_data["encryption"].get("storageClassName")
+        if encryption_storage_class_name:
+            data["encryption"] = encryption_storage_class_name
+
+    return data
+
+
+def check_custom_storageclass_presence():
+    """
+    Verify if the custom-defined storage class names are present in the `oc get sc` output.
+
+    Raises:
+        ValueError: If any of the custom-defined storage class names is not found in the output.
+
+    Returns:
+        bool: Returns True if all custom-defined storage class names are present in the `oc get sc` output.
```

Comment:
the function doesn't return False. Better to return False instead of raising an exception (ValueError) and raise the exception externally to this function